### DISCUSSION TITLE
`--jobname` and `--mode` in filters

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#397] [Q&A] [answered]
@pgundlach.

I read there is a way to access variables in filters (through the `vars` table).

I thought I read it (but I’m probably wrong), is there no way to access `--jobname` and `--mode` in filters?

Many thanks for your help.

**Comment by pgundlach** [comment]
Perhaps the `_mode` variable is available in the lua filter? See https://doc.speedata.de/publisher/en/internalvariables/

**Reply by pr-apes** [reply]
Sorry, but I cannot find it. With the simplest module available:

``` lua
print(_mode)
```

When applied to the following layout (and `sp --mode abc --dummy`):

``` xml

<Layout xmlns="urn:speedata.de:2009/publisher/en"
  xmlns:sd="urn:speedata:2009/publisher/functions/en">

  <SetGrid width="1mm" height="1mm"/>

  <Record element="data">
	<Output>
	<Text>
	<Paragraph>
	<Value select="$_mode" />
	</Paragraph>
      </Text>
    </Output>
  </Record>

</Layout>
```

I get `nil` in command prompt and `abc` in the PDF document.

BTW, exactly the same results with `_jobname` and `--jobname abc`.

**Comment by pgundlach** [comment, marked as answer]
sorry, this is:

```
runtime.options.jobname
runtime.options.mode
```

**Reply by pr-apes** [reply]
Many thanks for your reply, @pgundlach.
